src/runner.ts

```diff
@@ -173,25 +185,80 @@ async function runLinter(options: Options, logger: Logger): Promise<LintResult>
     return doLinting(options, files, program, logger);
 }
 
-function resolveFilesAndProgram(
-    { files, project, exclude, outputAbsolutePaths }: Options,
+/** Read the stdin stream into a string with all of the contents. */
+async function readStdin(): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+        let fileData = "";
+
+        process.stdin.setEncoding("utf8");
+
+        process.stdin.on("readable", () => {
+          let chunk: string | Buffer | null = null;
+
+          // tslint:disable-next-line: no-conditional-assignment
+          while ((chunk = process.stdin.read()) !== null) {
+            fileData += chunk;
+          }
+        });
+
+        process.stdin.on("end", () => {
+          resolve(fileData);
+        });
+
+        process.stdin.on("error", (error: Error) => {
+          reject(error);
+        });
```

Comment:
indentation is wrong throughout this file. some 4 spaces, some 2 spaces.

Reply:
I'll fix that when I get the time. I haven't managed to get tslint running on tslint code itself yet.